src/wasm-binary.h

```diff
@@ -76,24 +79,37 @@ struct LEB {
     } while (more);
   }
 
-  void read(std::function<MiniT ()> get) {
+  void read(std::function<MiniT()> get) {
     value = 0;
     T shift = 0;
     MiniT byte;
     while (1) {
       byte = get();
-      value |= ((T(byte & 127)) << shift);
-      if (!(byte & 128)) break;
+      bool last = !(byte & 128);
+      T payload = byte & 127;
+      typedef typename std::make_unsigned<T>::type mask_type;
```

Comment:
although I see this error exists later down in the code too... so can leave for now i guess